src/citrine/_utils/functions.py

```diff
@@ -319,3 +322,15 @@ def resource_path(*,
     new_url = base._replace(path='/'.join(path), query=query).geturl()
 
     return format_escaped_url(new_url, *action, **kwargs, uid=uid)
+
+def _data_manager_deprecation_checks(session, project_id:UUID, team_id:UUID, obj_type:str):
+    if project_id is None and team_id is None:
+        raise TypeError("A team_id must be provided.")
+    elif project_id is not None:
+        warn(f"{obj_type} now belong to Teams and not Projects. Providing a project_id/accessing via a project is deprecated and will be removed in future versions.",
+            DeprecationWarning
+        )
+        if team_id is None:
+            from citrine.resources.project import Project
```

Comment:
For ease of management, we import at the top of the module unless we cannot for some reason.

Reply:
It was a circular import otherwise :(

Reply:
Should I add an inline comment explaining that?